File: Darts/src/code/bean/GameParamFilterPanelGolf.java
package code.bean;
import java.awt.BorderLayout;
import java.awt.event.ActionListener;

import javax.swing.JRadioButton;

import bean.RadioButtonPanel;

public final class GameParamFilterPanelGolf extends GameParamFilterPanel
{
	public GameParamFilterPanelGolf()
	{
		super();
		
		add(panel, BorderLayout.CENTER);
		panel.add(rdbtn9);
		panel.add(rdbtn18);
		
		rdbtn18.setSelected(true); //Default to 18
	}
	
	private final RadioButtonPanel panel = new RadioButtonPanel();
	private final JRadioButton rdbtn9 = new JRadioButton("9 holes");
	private final JRadioButton rdbtn18 = new JRadioButton("18 holes");

	@Override
	public String getGameParams()
	{
		String selection = panel.getSelectionStr();
		return selection.replace(" holes", "");
	}
	@Override
	public void setGameParams(String gameParams)
	{
		if (gameParams.equals("9"))
		{
			rdbtn9.setSelected(true);
		}
		else
		{
			rdbtn18.setSelected(true);
		}
	}
	
	@Override
	public String getFilterDesc()
	{
		return "games of " + panel.getSelectionStr();
	}
	
	@Override
	public void enableChildren(boolean enabled)
	{
		rdbtn9.setEnabled(enabled);
		rdbtn18.setEnabled(enabled);
	}
	
	@Override
	public void addActionListener(ActionListener listener)
	{
		panel.addActionListener(listener);
	}
}

File: src/main/kotlin/dartzee/bean/GameParamFilterPanelGolf.kt
package dartzee.bean

import dartzee.core.bean.RadioButtonPanel
import java.awt.BorderLayout
import java.awt.event.ActionListener
import javax.swing.JRadioButton

class GameParamFilterPanelGolf : GameParamFilterPanel() {
    private val panel = RadioButtonPanel()
    val rdbtn9 = JRadioButton("9 holes")
    val rdbtn18 = JRadioButton("18 holes")

    init {

        add(panel, BorderLayout.CENTER)
        panel.add(rdbtn9)
        panel.add(rdbtn18)

        rdbtn18.isSelected = true // Default to 18
    }

    override fun getGameParams(): String {
        val selection = panel.getSelectionStr()
        return selection.replace(" holes", "")
    }

    override fun setGameParams(gameParams: String) {
        if (gameParams == "9") {
            rdbtn9.isSelected = true
        } else {
            rdbtn18.isSelected = true
        }
    }

    override fun getFilterDesc() = "games of ${panel.getSelectionStr()}"

    override fun enableChildren(enabled: Boolean) {
        rdbtn9.isEnabled = enabled
        rdbtn18.isEnabled = enabled
    }

    override fun addActionListener(listener: ActionListener) {
        panel.addActionListener(listener)
    }

    override fun removeActionListener(listener: ActionListener) {
        panel.removeActionListener(listener)
    }
}
